src/main/java/seedu/address/logic/commands/AddDoctorCommand.java

```diff
@@ -55,11 +55,11 @@ public AddDoctorCommand(Doctor doctor) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        if (model.hasPerson(doctorToAdd)) {
+        if (model.hasDoctor(doctorToAdd)) {
```

Comment:
Great use of inheritance to extend the functionality of the Person class through its child classes

Reply:
Applies to basically all the changes made in this PR
Awesome job being thorough and making all the relevant changes/updates!